src/state.rs

```diff
@@ -698,18 +706,37 @@ impl DemandProxyState {
     ) -> Result<Option<Upstream>, Error> {
         self.fetch_address(&network_addr(network.clone(), addr.ip()))
             .await;
-        let Some((wl, port, svc)) = self.state.read().unwrap().find_upstream(
-            network,
-            source_workload,
-            addr,
-            resolution_mode,
-        ) else {
+        let upstream = {
+            self.state.read().unwrap().find_upstream(
```

Comment:
**nit**: expect instead of unwrap

Reply:
(can we just make clippy flag `unwrap`? might not be terrible)

Reply:
quick bit of work on this, we have something in the range of 60 uses of unwrap outside of tests and build stuff so non-trivial to just yeet it. Happy to do the work to cleanup and enable the lint if we want to go this route though.

Reply:
https://github.com/istio/ztunnel/issues/1245